demo-shell/src/app/components/app-layout/user-info/user-info.component.ts

```diff
@@ -52,23 +34,12 @@ export class UserInfoComponent implements OnInit, OnDestroy {
     @Input()
     menuPositionY: MenuPositionY = 'below';
 
-    /** Shows/hides the username next to the user info button. */
-    @Input()
-    showName: boolean = true;
-
-    /** When the username is shown, this defines its position relative to the user info button.
-     * Can be `right` or `left`.
-     */
-    @Input()
-    namePosition: string = 'right';
-
-    mode: string;
-
+    mode: UserInfoMode;
     ecmUser$: Observable<EcmUserModel>;
     bpmUser$: Observable<BpmUserModel>;
     identityUser$: Observable<IdentityUserModel>;
     selectedIndex: number;
-    private destroy$ = new Subject();
+    userInfoMode = UserInfoMode;
 
     constructor(private peopleContentService: PeopleContentService,
```

Comment:
user-info.component.ts should not depend from  peopleContentService and bpmUserService to be generic